torch/csrc/distributed/c10d/ProcessGroup.hpp

```diff
@@ -59,10 +59,11 @@ class TORCH_API ProcessGroup : public torch::CustomClassHolder {
     std::chrono::milliseconds timeout;
 
     // backend name
+    // NOLINTNEXTLINE(cppcoreguidelines-avoid-const-or-ref-data-members)
     const std::string backend;
   };
 
-  enum BackendType {
+  enum BackendType : uint8_t {
```

Comment:
Hmm, what clang-tidy rule is that?

Reply:
The newer clang-tidy 18 has added performance-enum-size to warn enum size, see
https://clang.llvm.org/extra/clang-tidy/checks/performance/enum-size.html